lib/puppet/type/esx_vswitch.rb

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) 2013 VMware, Inc.
+Puppet::Type.newtype(:esx_vswitch) do
+  @doc = "Manage vCenter vSwitch."
+
+  ensurable do
+    newvalue(:present) do
+      provider.create
+    end
+    newvalue(:absent) do
+      provider.destroy
+    end
+    defaultto(:present)
+  end
+
+  newparam(:name, :namevar => true) do
+    desc "ESX host:vSwitch name."

```

Comment:
@maniacmurphy:  I agree - and isn't this better done with composite namevars and title_patterns?.... I'll fix this up and submit a new PR for it